tests/unit/test_spaces.cpp

```diff
@@ -163,6 +164,150 @@ static dist_func_t<double> IP_dist_funcs_2ExtResiduals[] = {
     FP64_InnerProductSIMD2ExtResiduals_AVX512};
 } // namespace spaces_test
 
+TEST_F(SpacesTest, fp32_L2_GetDistFunc) {
+    using namespace spaces_test;
+    using namespace spaces;
+    // fp32_L2_dist_funcs_array[i][j] is a pointer to a function that match CalculationGuideline[i]
+    // and architecture optimization Arch_Optimization[j]
+    static dist_func_t<float> *fp32_L2_dist_funcs_array[] = {
+        nullptr, L2_dist_funcs_16Ext, L2_dist_funcs_4Ext, L2_dist_funcs_16ExtResiduals,
+        L2_dist_funcs_4ExtResiduals};
+    static size_t dimensions[] = {
+        3,  // NO_OPTIMIZATION
+        16, // % 16 ==0,
+        20, //% 4 == 0,
+        17, //>16, % 16 < 4 ,
+        5,  //> 4  ]
+    };
+    /* enum Arch_Optimization {
+    ARCH_OPT_NONE,
+    ARCH_OPT_SSE,
+    ARCH_OPT_AVX,
+    ARCH_OPT_AVX512_F,
+    ARCH_OPT_AVX512_DQ // Relevant only to fp64
+}; */
+    // For each arch
+    for (int arch = ARCH_OPT_NONE; arch <= ARCH_OPT_AVX512_F; ++arch) {
+        ASSERT_EQ(L2_FP32_GetDistFunc(dimensions[NO_OPTIMIZATION], (Arch_Optimization)arch),
```

Comment:
can you add this logic to the loop?

Reply:
let's say we remove the nullptr and the inner loop begins with dim_opt = NO_OPTIMIZATION
then the first entire loop is ok,
but in the second iteration (arch = 1) we get 
`ASSERT_EQ(L2_FP32_GetDistFunc(dimensions[0], (Arch_Optimization)1), //should be fp32_L2Sqr
                      fp32_L2_dist_funcs_array[0][1] // SSE`